pkg/auth/handler/login.go

```diff
@@ -89,6 +91,24 @@ func (h LoginHandler) DecodeRequest(request *http.Request) (handler.RequestPaylo
 // Handle api request
 func (h LoginHandler) Handle(req interface{}, ctx context.AuthContext) (resp interface{}, err error) {
 	payload := req.(LoginRequestPayload)
+	fetchedAuthInfo := authinfo.AuthInfo{}
+
+	defer func() {
+		if h.AuditTrail == nil {
+			return
+		}
```

Comment:
Let keep the original design of auditTrail that it is not optional. It always exist, but there is no hook/set to panic level if the Tenant does not config any.